crates/virtio-queue/src/descriptor_utils.rs

```diff
@@ -0,0 +1,939 @@
+// Copyright 2019 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+use std::collections::VecDeque;
+use std::fmt::{self, Display};
+use std::io::{self, Read, Write};
+use std::mem::{size_of, MaybeUninit};
+use std::ops::Deref;
+use std::ptr::copy_nonoverlapping;
+use std::{cmp, result};
+
+use crate::DescriptorChain;
+use vm_memory::bitmap::{Bitmap, BitmapSlice};
+use vm_memory::{
+    Address, ByteValued, GuestMemory, GuestMemoryError, GuestMemoryMmap, GuestMemoryRegion, Le16,
+    Le32, Le64, VolatileMemory, VolatileMemoryError, VolatileSlice,
+};
+
+#[derive(Debug)]
+pub enum Error {
+    DescriptorChainOverflow,
+    FindMemoryRegion,
+    GuestMemoryError(GuestMemoryError),
+    InvalidChain,
+    IoError(io::Error),
+    SplitOutOfBounds(usize),
+    VolatileMemoryError(VolatileMemoryError),
+}
```

Comment:
The rest of the crate seems to mostly use `crate::Error`. Can we reuse that here too (maybe by adding an error or two to the crate-wide Error enum?)